pkg/controller/apiserver/apiserver_controller.go

```diff
@@ -36,28 +30,37 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/controller/installation"
+	"github.com/tigera/operator/pkg/controller/k8sapi"
+	"github.com/tigera/operator/pkg/controller/options"
+	"github.com/tigera/operator/pkg/controller/status"
+	"github.com/tigera/operator/pkg/controller/utils"
+	"github.com/tigera/operator/pkg/render"
 )
 
 var log = logf.Log.WithName("controller_apiserver")
 
 // Add creates a new APIServer Controller and adds it to the Manager. The Manager will set fields on the Controller
 // and Start it when the Manager is Started.
-func Add(mgr manager.Manager, opts options.AddOptions) error {
+func Add(k8sServiceEp k8sapi.ServiceEndpoint, mgr manager.Manager, opts options.AddOptions) error {
```

Comment:
I don't think there is any need to pass the k8sServiceEp through here. We didn't in the core_controller.

I'm half tempted to suggest we don't even pass it into the render functions and just have the renders use it directly. But I think I like it as an input the renders, I think...

Reply:
I like having it as an input to rendering; makes it easier to test.